### DISCUSSION TITLE
Is RTL Supported?

### DISCUSSION THREAD

**Discussion by omarhamid89** [mearashadowfax/ScrewFast#121] [Q&A] [answered]
I apologize if this is obvious but I'm new to astro.

I want to use ScrewFast for a project but need it to support Arabic, are there any plans to support Right to Left text in the future?

If it is indeed supported, then I'm struggling with making it work. I would appreciate it if you point me in the right direction.

Thank you

**Comment by mearashadowfax** [comment, marked as answer]
Hello @omarhamid89! Tailwind does provide [RTL support](https://tailwindcss.com/docs/hover-focus-and-other-states#rtl-support). If your site has a single direction, you just need to add `dir="rtl"` to the `<html>` tag in the `MainLayout.astro` component: `<html dir="rtl" lang={lang} class="scrollbar-hide lenis lenis-smooth scroll-pt-16">`.

This will enable RTL support across your site. However, you might want to make some adjustments too, like swapping the border radius from the left to the right for certain sections, such as the pricing section, etc.

![localhost_4321_](https://github.com/mearashadowfax/ScrewFast/assets/125820963/245981ee-27fa-4c20-9f07-343275d23327)
![localhost_4321_ (1)](https://github.com/mearashadowfax/ScrewFast/assets/125820963/313cc1d2-ce3f-4190-8104-6d568a7b5dce)

**Reply by omarhamid89** [reply]
Thank you, this is what I've been looking for!
I'm planning a bilingual site and I think this will affect the English version too, correct?
If yes, is there any way to just enable it for the RTL language?

**Reply by mearashadowfax** [reply]
Yes, so for the English version you will have `<html dir="ltr">`. Then, using the `LanguagePicker.astro` component, you can adjust it to change the `dir` attribute of the `<html>` element when the language changes. Something like this:

```js
// src/components/ui/LanguagePicker.astro
const languageDirection: { [key in TLanguage]: "ltr" | "rtl" } = {
    en: "ltr",
    fr: "ltr",
    ar: "rtl",
  };

// existing code

document.documentElement.setAttribute("dir", languageDirection[lang]);
```
